CRM/Mailing/BAO/MailingJob.php

```diff
@@ -520,7 +520,8 @@ public function deliver(&$mailer, $testParams = NULL) {
                         AND     $edTable.id IS null
                         AND     $ebTable.id IS null
                         AND    contact_a.is_opt_out = 0 
-                        $aclWhere";
+                        $aclWhere
+                    ORDER BY    $contactTable.contact_type";
```

Comment:
Jenkins test failure seems related. Should be fixed by using the alias(contact_a) instead of `$contactTable`.